Update TCKResultsWriter to generate a separate file for each repeat

jakub-pomykala opened the pull request:
- TCKResultsWriter now creates a separate report file per each repeat
- The test names no longer have the repeat ID appended to them

Fixes #27169

jakub-pomykala committed:
Update TCKResultsWriter to generate separate files

- TCKResultsWriter now creates a separate file per each repeat

jakub-pomykala committed:
Adding in comments

tevans78 commented on dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsWriter.java:
```diff
@@ -36,7 +36,7 @@ public class TCKResultsWriter {
      * @param resultInfo TCK Results metadata
      */
-    public static void preparePublicationFile(TCKResultsInfo resultInfo) {
+    public static void preparePublicationFile(TCKResultsInfo resultInfo, String repeat) {
```
please update the javadoc too

tevans78 replied:
be clear if a null repeat string is allowed/expected

tevans78 replied:
instead of adding the repeat param here, it might be better to include the repeat string as part of `TCKResultsInfo`

jakub-pomykala replied:
Moved to `TCKResultsInfo`.

A `null` will never be returned from the [RepeatTestFilter.getRepeatActionsAsString() ](https://github.com/OpenLiberty/open-liberty/blob/d1f0e4e8064d91deef53dfc06700ea667d0bd4ee/dev/fattest.simplicity/src/componenttest/custom/junit/runner/RepeatTestFilter.java#L151) because if the `REPEAT_ACTION_STACK` is empty the string returned is set to be`""`.

tevans78 commented on dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsWriter.java:
```diff
@@ -74,7 +74,7 @@ public static void preparePublicationFile(TCKResultsInfo resultInfo) {
             tckURL = "https://download.eclipse.org/ee4j/" + specName + "/jakartaee10/promoted/eftl/" + specName + "-tck-" + specVersion + ".zip"; //just a placeholder, needs to be manually updated
         }
-        String filename = openLibertyVersion + "-" + fullSpecName.replace(" ", "-") + "-Java" + javaMajorVersion + "-TCKResults.adoc";
+        String filename = openLibertyVersion + "-" + fullSpecName.replace(" ", "-") + "-Java" + javaMajorVersion + "-TCKResults" + repeat + ".adoc";
```
Does the repeat string already include a separator char? Should a `-` be added after `-TCKResults`. What happens if the repeat string is null?

jakub-pomykala replied:
Yes it includes `_` . But in the new commit I have made it so all the `_` are replaced with `-` to keep it consistent with the `filenames`.

tevans78 commented on dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKResultsWriter.java:
```diff
@@ -94,7 +94,12 @@ public static void preparePublicationFile(TCKResultsInfo resultInfo) {
             for (TestSuiteResult result : xmlParser.getResults()) {
-                output.write(result.toString());
+                // Checking if the result is for the correct repeat (avoiding adding wrong repeat results to wrong TCK report repeat files)
+                if (result.toString().contains(repeat)) {
```
do you want a null check on the repeat string?

tevans78 replied:
does the code which adds the repeat string ever include a separator char after the test name?

jakub-pomykala replied:
answered in https://github.com/OpenLiberty/open-liberty/pull/27429#pullrequestreview-1841044600

tevans78 commented on dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKRunner.java:
```diff
@@ -258,7 +258,8 @@ private void runTCK() throws Exception {
         TCKJarInfo tckJarInfo = TCKUtilities.getTCKJarInfo(this.type, dependencyOutput);
         TCKResultsInfo resultsInfo = new TCKResultsInfo(this.type, this.specName, this.server, tckJarInfo);
-        TCKResultsWriter.preparePublicationFile(resultsInfo);
+        String repeat = RepeatTestFilter.getRepeatActionsAsString();
```
in theory, it is possible to have multiple repeat actions active at once. Not sure what would happen then.

jakub-pomykala committed:
Moving repeat string to TCKResultsInfo

- Passing the repeat string to TCKResultsWriter from TCKRunner in the TCK ResultsInfo instead of separate.
- Updating comments
- Adding the repeat string to JavaDoc

jakub-pomykala committed:
Updating copyright date in RepeatTestFilter

jakub-pomykala committed:
Revert "Updating copyright date in RepeatTestFilter"

This reverts commit fbff4709848030f2102df360d57b1170badfa788.

jakub-pomykala committed:
Updating copyright date in TCKResultsInfo

jakub-pomykala committed:
Fixing TCKReports not being uploaded

- TCK Reports need to end with "TCKResults.adoc" to be correctly uploaded

jakub-pomykala commented:
#build
#spawn.fullfat.buckets=com.ibm.ws.concurrent.mp.1.2_fat_tck,com.ibm.ws.concurrent.mp.1.3_fat_tck,com.ibm.ws.concurrent.mp_fat_tck,com.ibm.ws.microprofile.config.1.1_fat_tck,com.ibm.ws.microprofile.config.1.2_fat_tck,com.ibm.ws.microprofile.config.1.3_fat_tck,com.ibm.ws.microprofile.config.1.4_fat_tck,com.ibm.ws.microprofile.faulttolerance.1.1_fat_tck,com.ibm.ws.microprofile.faulttolerance.2.0_fat_tck,com.ibm.ws.microprofile.faulttolerance.2.1_fat_tck,com.ibm.ws.microprofile.faulttolerance_fat_tck,com.ibm.ws.microprofile.graphql_fat_tck,com.ibm.ws.microprofile.health.1.0_fat_tck,com.ibm.ws.microprofile.health.2.0_fat_tck,com.ibm.ws.microprofile.health.2.1_fat_tck,com.ibm.ws.microprofile.health.2.2_fat_tck,com.ibm.ws.microprofile.metrics.1.1.noAuth_fat_tck,com.ibm.ws.microprofile.metrics.1.1_fat_tck,com.ibm.ws.microprofile.metrics.2.0_fat_tck,com.ibm.ws.microprofile.metrics.2.2_fat_tck,com.ibm.ws.microprofile.metrics.2.3_fat_tck,com.ibm.ws.microprofile.metrics_fat_tck,com.ibm.ws.microprofile.mpjwt.1.1_fat_tck,com.ibm.ws.microprofile.openapi.1.1_fat_tck,com.ibm.ws.microprofile.openapi_fat_tck,com.ibm.ws.microprofile.opentracing.1.1_fat_tck,com.ibm.ws.microprofile.opentracing.1.2_fat_tck,com.ibm.ws.microprofile.opentracing.1.3_fat_tck,com.ibm.ws.microprofile.opentracing_fat_tck,com.ibm.ws.microprofile.reactive.messaging_fat_tck,com.ibm.ws.microprofile.reactive.streams.operators_fat_tck,com.ibm.ws.microprofile.rest.client11_fat_tck,com.ibm.ws.microprofile.rest.client12_fat_tck,com.ibm.ws.microprofile.rest.client13_fat_tck,com.ibm.ws.microprofile.rest.client14_fat_tck,com.ibm.ws.microprofile.rest.client_fat_tck,com.ibm.ws.security.mp.jwt_fat_tck,io.openliberty.jakarta.concurrency.3.0_fat_tck,io.openliberty.jakarta.concurrency.3.1_fat_tck,io.openliberty.jakarta.data.1.0_fat_tck,io.openliberty.jakarta.jsonb.3.0_fat_tck,io.openliberty.jakarta.jsonp.2.1_fat_tck,io.openliberty.jakarta.rest.3.1.internal_fat_tck,io.openliberty.microprofile.config.2.0.internal_fat_tck,io.openliberty.microprofile.config.3.0.internal_fat_tck,io.openliberty.microprofile.config.3.1.internal_fat_tck,io.openliberty.microprofile.faulttolerance.3.0.internal_fat_tck,io.openliberty.microprofile.faulttolerance.4.0.internal_fat_tck,io.openliberty.microprofile.graphql.2.0.internal_fat_tck,io.openliberty.microprofile.health.3.0.internal_fat_tck,io.openliberty.microprofile.health.3.1.internal_fat_tck,io.openliberty.microprofile.health.4.0.internal_fat_tck,io.openliberty.microprofile.jwt.1.2.internal_fat_tck,io.openliberty.microprofile.jwt.2.0.internal_fat_tck,io.openliberty.microprofile.jwt.2.1.internal_fat_tck,io.openliberty.microprofile.lra.1.0.internal_fat_tck,io.openliberty.microprofile.metrics.internal.3.0_fat_tck,io.openliberty.microprofile.metrics.internal.4.0_fat_tck,io.openliberty.microprofile.metrics.internal.5.0_fat_tck,io.openliberty.microprofile.metrics.internal.5.1_fat_tck,io.openliberty.microprofile.openapi.2.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.1.internal_fat_tck,io.openliberty.microprofile.opentracing.2.0.internal_fat_tck,io.openliberty.microprofile.opentracing.3.0.internal_fat_tck,io.openliberty.microprofile.reactive.messaging30.internal_fat_tck,io.openliberty.microprofile.reactive.streams.operators30.internal_fat_tck,io.openliberty.microprofile.rest.client.2.0.internal_fat_tck,io.openliberty.microprofile.rest.client.3.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

jakub-pomykala committed:
Adding featureReplecementAction check

- If there is a repeat which does not have an ID it will be substituted with "REMOVE `OLD_FEATURE_NAME` ADD `NEW_FEATURE_NAME` this will break the TCK report name as that is an incorrect repeat name therefore doing this check and also  replacing it with "...remove_OLD_FEATURE_NAME_add_NEW_FEATURE_NAME"

jakub-pomykala commented:
#build
#spawn.fullfat.buckets=com.ibm.ws.concurrent.mp.1.2_fat_tck,com.ibm.ws.concurrent.mp.1.3_fat_tck,com.ibm.ws.concurrent.mp_fat_tck,com.ibm.ws.microprofile.config.1.1_fat_tck,com.ibm.ws.microprofile.config.1.2_fat_tck,com.ibm.ws.microprofile.config.1.3_fat_tck,com.ibm.ws.microprofile.config.1.4_fat_tck,com.ibm.ws.microprofile.faulttolerance.1.1_fat_tck,com.ibm.ws.microprofile.faulttolerance.2.0_fat_tck,com.ibm.ws.microprofile.faulttolerance.2.1_fat_tck,com.ibm.ws.microprofile.faulttolerance_fat_tck,com.ibm.ws.microprofile.graphql_fat_tck,com.ibm.ws.microprofile.health.1.0_fat_tck,com.ibm.ws.microprofile.health.2.0_fat_tck,com.ibm.ws.microprofile.health.2.1_fat_tck,com.ibm.ws.microprofile.health.2.2_fat_tck,com.ibm.ws.microprofile.metrics.1.1.noAuth_fat_tck,com.ibm.ws.microprofile.metrics.1.1_fat_tck,com.ibm.ws.microprofile.metrics.2.0_fat_tck,com.ibm.ws.microprofile.metrics.2.2_fat_tck,com.ibm.ws.microprofile.metrics.2.3_fat_tck,com.ibm.ws.microprofile.metrics_fat_tck,com.ibm.ws.microprofile.mpjwt.1.1_fat_tck,com.ibm.ws.microprofile.openapi.1.1_fat_tck,com.ibm.ws.microprofile.openapi_fat_tck,com.ibm.ws.microprofile.opentracing.1.1_fat_tck,com.ibm.ws.microprofile.opentracing.1.2_fat_tck,com.ibm.ws.microprofile.opentracing.1.3_fat_tck,com.ibm.ws.microprofile.opentracing_fat_tck,com.ibm.ws.microprofile.reactive.messaging_fat_tck,com.ibm.ws.microprofile.reactive.streams.operators_fat_tck,com.ibm.ws.microprofile.rest.client11_fat_tck,com.ibm.ws.microprofile.rest.client12_fat_tck,com.ibm.ws.microprofile.rest.client13_fat_tck,com.ibm.ws.microprofile.rest.client14_fat_tck,com.ibm.ws.microprofile.rest.client_fat_tck,com.ibm.ws.security.mp.jwt_fat_tck,io.openliberty.jakarta.concurrency.3.0_fat_tck,io.openliberty.jakarta.concurrency.3.1_fat_tck,io.openliberty.jakarta.data.1.0_fat_tck,io.openliberty.jakarta.jsonb.3.0_fat_tck,io.openliberty.jakarta.jsonp.2.1_fat_tck,io.openliberty.jakarta.rest.3.1.internal_fat_tck,io.openliberty.microprofile.config.2.0.internal_fat_tck,io.openliberty.microprofile.config.3.0.internal_fat_tck,io.openliberty.microprofile.config.3.1.internal_fat_tck,io.openliberty.microprofile.faulttolerance.3.0.internal_fat_tck,io.openliberty.microprofile.faulttolerance.4.0.internal_fat_tck,io.openliberty.microprofile.graphql.2.0.internal_fat_tck,io.openliberty.microprofile.health.3.0.internal_fat_tck,io.openliberty.microprofile.health.3.1.internal_fat_tck,io.openliberty.microprofile.health.4.0.internal_fat_tck,io.openliberty.microprofile.jwt.1.2.internal_fat_tck,io.openliberty.microprofile.jwt.2.0.internal_fat_tck,io.openliberty.microprofile.jwt.2.1.internal_fat_tck,io.openliberty.microprofile.lra.1.0.internal_fat_tck,io.openliberty.microprofile.metrics.internal.3.0_fat_tck,io.openliberty.microprofile.metrics.internal.4.0_fat_tck,io.openliberty.microprofile.metrics.internal.5.0_fat_tck,io.openliberty.microprofile.metrics.internal.5.1_fat_tck,io.openliberty.microprofile.openapi.2.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.0.internal_fat_tck,io.openliberty.microprofile.openapi.3.1.internal_fat_tck,io.openliberty.microprofile.opentracing.2.0.internal_fat_tck,io.openliberty.microprofile.opentracing.3.0.internal_fat_tck,io.openliberty.microprofile.reactive.messaging30.internal_fat_tck,io.openliberty.microprofile.reactive.streams.operators30.internal_fat_tck,io.openliberty.microprofile.rest.client.2.0.internal_fat_tck,io.openliberty.microprofile.rest.client.3.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

jakub-pomykala commented:
Test failure and error in PR build are not related to the changes in this PR. Generated TCK Reports look as expected.

jakub-pomykala commented:
#libby
